packages/components/src/components/dropdown/dropdown.jsx

```diff
@@ -265,6 +266,12 @@ const Dropdown = ({
     };
 
     const handleVisibility = () => {
+        if (typeof onClick === 'function') {
+            onClick();
+
+            return;
+        }
+
         if (is_nativepicker && !is_list_visible) {
```

Comment:
`nit`, maybe something like this could work, like:
```
if (typeof onClick === 'function') {
   onClick()
} else {
   if (is_nativepicker && !is_list_visible) {
      if (mobileOSDetect() === 'iOS') {
                /* .focus() doesn't trigger open <select /> in Android :(
                 * so we use a CSS hack - refer to L237 in dropdown.scss
                 * [TODO]: find alternative solution to trigger open <select /> with JS
                 */
                native_select_ref.current.focus();
            }
      setIsListVisible(true);
   } else {
       setIsListVisible(!is_list_visible);
   }
}
```